packages/testlab/src/misc.ts

```diff
@@ -0,0 +1,16 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/testlab
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+/**
+ * JSON transport does not preserve properties that are undefined
+ * As a result, deepEqual checks fail because the expected model
+ * value contains these undefined property values, while the actual
+ * result returned by REST API does not.
+ * Use this function to convert a model instance into a data object
+ * as returned by REST API
+ */
+export function deserializedFromJson<T extends object>(value: T): T {
```

Comment:
The method name is a bit misleading. It tricks me to think about the argument is a JSON string.

What about `asJSON()` or `toJSON()`?

Do we want to restrain the argument to be an `object`? What about `any`?

We need to deal with the case where value is `undefined`. 
```ts
JSON.parse(JSON.stringify(undefined)); // SyntaxError: Unexpected token u in JSON at position 0
```
The return type is incorrect, for example:

```ts
// Customer is a class with `getFullName()` method.
const customer: Customer = await this.repo.findById('c001');
const json = deserializedFromJson(customer); // TypeScript infers the return type as `Customer`
json.getFullName(); // The function does not exist but TypeScript cannot catch it.
```

I propose that we change the method to be:
```
export function toJSON<T extends object>(value: T): DeepPartial<T> { // ... }
```

Reply:
+1 to using either `asJSON` or `toJSON` ... with a slight preference for `toJSON`. 

```
// Customer is a class with `getFullName()` method.
const customer: Customer = await this.repo.findById('c001');
const json = deserializedFromJson(customer); // TypeScript infers the return type as `Customer`
json.getFullName(); // The function does not exist but TypeScript cannot catch it.
```

I don't think `json.getFullName();` is a valid use of this as the intention of this method is to act as a helper to allow comparison of an object to the value you would get from a rest api ... which is the use case for testing (and the inspiration for this helper method). One should just do `customer.getFullName()` instead of converting to JSON and then trying to treat it as a class. 

Considering the above maybe a better name we can use is `modelToJSON`?

Reply:
@virkt25 Please note by the current signature, `deserializedFromJson(customer); ` returns a `Customer`, which has `getFullName()`. As a result, TypeScript thinks `json.getFullName()` is valid use but the `getFullName` does not exist on the object.